helix-core/src/main/java/org/apache/helix/controller/stages/BestPossibleStateCalcStage.java

```diff
@@ -104,7 +109,7 @@ public Object call() {
   }
 
   private BestPossibleStateOutput compute(ClusterEvent event, Map<String, Resource> resourceMap,
-      CurrentStateOutput currentStateOutput) {
+      CurrentStateOutput currentStateOutput) throws JMException {
```

Comment:
Same here, we don't want any monitor logic to block business. No matter how broken it could be.